controllers/statefulset.go

```diff
@@ -870,6 +925,16 @@ func (r *SingleClusterReconciler) updateSTSSchedulingPolicy(
 		)
 	}
 
+	if r.aeroCluster.Spec.K8sNodeBlockList != nil {
```

Comment:
`K8sNodeBlockList` is omitempty, so if empty is given by user, it would be skipped (nil).
Both are similar checks here. Will change to len check. 
Also will add a kub-builder min-length validation.